Plugin&maven

frenzqse opened the pull request:
Added poms, to generate the signaldomain dlls

frenzqse committed:
Add plugin

frenzqse committed:
[OPENENGSB-2721] Add plugin

frenzqse committed:
[OPENENGSB-2721] Add plugin to the maven dependency and let generate the dlls

frenzqse committed:
Format poms

frenzqse committed:
Add assembly.cs

frenzqse committed:
[OPENENGSB-2686] Add binary distribution

frenzqse committed:
Change name of the wsdl to dll poms

anpieber commented:
mhm, something is wrong here again... this pull request can not be merged automatically... I'm afraid there's something wrong with  the branch versions again...

anpieber commented on assembly/pom.xml:
```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="utf-8"?>
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns="http://maven.apache.org/POM/4.0.0">
```
this file is a mess in formatting (tabs/spaces). Please check again...

anpieber commented on bridge/Implementation/pom.xml:
```diff
@@ -93,14 +87,14 @@
-      <groupId>org.openengsb.wrapped.csharp</groupId>
-      <artifactId>Newtonsoft.Json</artifactId>
-      <version>5.0_1-SNAPSHOT</version>
+      <groupId>org.openengsb.loom.csharp.common.bridge</groupId>
```
IIRC those references had been fixed already on the master

frenzqse replied:
The pull requests are still open. Did you merged them?

anpieber replied:
give the resulting files a look; I think so. maybe I forget to close them :-)

frenzqse replied:
Oh I just saw it. They are merged. Are the new version on maven central as well?

anpieber replied:
nope (since they're still snapshots). I'll release them once we decide to release this package. Though, if you want I can also release them now

anpieber commented on bridge/ServiceTestConsole/Program.cs:
```diff
@@ -40,7 +40,7 @@ static void Main(string[] args)
             string domainName = "signal";
 
-            IDomainFactory factory = DomainFactoryProvider.GetDomainFactoryInstance(new Uri("http://localhost:8090"));
+            IDomainFactory factory = DomainFactoryProvider.GetDomainFactoryInstance("3.0.0");
```
3.0.0? shouldn't it be rather 3.x? or 3.0

anpieber replied:
micro releases will NEVER EVER change anything in the api.... minor releases could maybe add additional features

frenzqse replied:
There are two ways to pick the correct version.
With a Url, which the program uses to get the version number.
or by entering the version number.
Actually it doesn't matter if you write there 3.0.0 or 3.9.3. The program is just considering the first number.
For testing reasons I used the version number, but for the release, I will change it back to url

anpieber replied:
what do you mean by URL?

frenzqse replied:
The Url, to get the OpenEngSB version, like vor example http://localhost:8090

anpieber replied:
ahhhhhh ok; yes that will be the better solution anyhow; you're right

anpieber commented on bridge/ServiceTestConsole/pom.xml:
```diff
@@ -42,21 +42,25 @@
     </dependency>
     <dependency>
-      <groupId>org.openengsb.wrapped.csharp</groupId>
+      <groupId>SignalDomainInterfaces</groupId>
```
groupd id should be unique everywhere on the world ... the old one was WAYS better

frenzqse replied:
I had some problem, whiph including the dll ins Vs or in npanday. I don't remember exactly the problem, but it had to do with integration of the dll in VS and npanday. And one of both couldn't find them with that group Id. The way, which leads to no problems, was this one.

anpieber replied:
please check this again; this group id is a real problem for someone used to work with mvn

frenzqse replied:
Ok. I will check, what I can do

anpieber commented on bridge/ServiceTestConsole/pom.xml:
```diff
@@ -42,21 +42,25 @@
-      <version>1.2.11_1-SNAPSHOT</version>
+      <groupId>SignalDomainEventsInterfaces</groupId>
+      <artifactId>SignalDomainEventsInterfaces</artifactId>
+      <version>0.0.0.0</version>
```
whats that for a version?

frenzqse replied:
Actually only the system path is relevant, because the dll is include relatively. Furthermore, the dll has no version nbr. because it is generated from one file. Maybe it is possible to give a version nmbr while compiling but it doesn't mater, because the dll is include over the systemPath

anpieber replied:
nevermind; use the same version as the project. It should update automatically then. 0.0.0 is simply never a good idea

frenzqse replied:
It is autogenerated from npanday. But I will change it to 1.0.0

anpieber replied:
thx

anpieber commented on bridge/WsdlToDll/SignalDomain/pom.xml:
```diff
@@ -0,0 +1,76 @@
+                <artifactItem>
+                  <groupId>at.ac.tuwien.ifs.cdl.domain</groupId>
+                  <artifactId>at.ac.tuwien.ifs.cdl.domain.signal</artifactId>
+				  <classifier>SignalDomain</classifier>				  
```
formatting

anpieber replied:
re-check the entire file; not only this line

anpieber commented on bridge/WsdlToDll/SignalDomainEvents/pom.xml:
```diff
@@ -0,0 +1,76 @@
+                <artifactItem>
+                  <groupId>at.ac.tuwien.ifs.cdl.domain</groupId>
+                  <artifactId>at.ac.tuwien.ifs.cdl.domain.signal</artifactId>
+				  <classifier>SignalDomainEvents</classifier>				  
```
again formatting

anpieber commented on bridge/WsdlToDll/pom.xml:
```diff
@@ -0,0 +1,16 @@
+  <name>OpenEngSB :: Loom :: CSharp :: Common :: Bridge :: Convert :: WSDL</name>
+  <modules>   
+    <module>SignalDomain</module>
+	<module>SignalDomainEvents</module>
```
formatting

anpieber commented on wsdltodll/src/main/java/generate/wsdl/WsdlToDll.java:
```diff
@@ -0,0 +1,189 @@
+ * 
+ * @phase process-sources
+ */
+public class WsdlToDll extends AbstractMojo {
```
pls use the openengsb eclipse formatter; from the "what" point of  view +1; from the "how" point of view; -1. I'll update it ASAP you correct your branch structure.

anpieber replied:
forget this command; I'll adapt and reformat the entire thing once it is merged

anpieber commented:
all in all it looks pretty fine; pls check two problems:

1) branch does not seam to be correct (base on master)
2) formatting of most pom files

pls ping me again here then. I'll adapt and test various other things then before merging

frenzqse committed:
format poms

frenzqse committed:
Merge branch 'master' into plugin&maven

Conflicts:
	assembly/pom.xml
	wsdltodll/pom.xml

frenzqse committed:
change version number of the external dll to the project version

frenzqse committed:
Merge remote-tracking branches 'origin' and 'cmaster/master' into plugin&maven

Conflicts:
	assembly/pom.xml
	bridge/Implementation/Communication/Jms/JmsOutgoingPort.cs
	bridge/Implementation/Implementation.csproj
	bridge/Implementation/OpenEngSB2_4_0/Remote/DomainProxy.cs
	bridge/Implementation/pom.xml
	bridge/ServiceTestConsole/Program.cs
	bridge/ServiceTestConsole/ServiceTestConsole.csproj
	bridge/ServiceTestConsole/pom.xml
	bridge/pom.xml
	docs/src/manual/docbook/install/configuration.xml
	docs/src/manual/loom-csharp-common-manual.xml

anpieber commented on assembly/descriptors/bundle.xml:
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
license is missing

anpieber commented on assembly/pom.xml:
```diff
@@ -14,8 +13,61 @@
 
   <name>OpenEngSB :: Loom :: CSharp :: Common :: Assembly</name>
-
-  <!-- Add the maven-assembly-plugin sections here -->
```
license is missing

anpieber commented on bridge/ServiceTestConsole/ServiceTestConsole.csproj:
```diff
@@ -38,11 +38,21 @@
     <Reference Include="log4net">
       <HintPath>.references\org.openengsb.wrapped.csharp\log4net-1.2.11_1-SNAPSHOT\log4net.dll</HintPath>
     </Reference>
+<<<<<<< .merge_file_a04200
```
BE CAREFUL!!! ERROR!!!!

anpieber commented on bridge/WsdlToDll/SignalDomain/pom.xml:
```diff
@@ -0,0 +1,76 @@
+              <artifactItems>
+                <artifactItem>
+                  <groupId>at.ac.tuwien.ifs.cdl.domain</groupId>
+                  <artifactId>at.ac.tuwien.ifs.cdl.domain.signal</artifactId>
```
remove the signal example... this one is a little bit hard for others to reproduce...

frenzqse replied:
I am working on it. It should be finished this afternoon

anpieber commented on bridge/WsdlToDll/SignalDomainEvents/pom.xml:
```diff
@@ -0,0 +1,76 @@
+              <artifactItems>
+                <artifactItem>
+                  <groupId>at.ac.tuwien.ifs.cdl.domain</groupId>
+                  <artifactId>at.ac.tuwien.ifs.cdl.domain.signal</artifactId>
```
other domain

anpieber commented on bridge/WsdlToDll/pom.xml:
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns="http://maven.apache.org/POM/4.0.0">
```
license

anpieber commented on bridge/pom.xml:
```diff
@@ -11,10 +11,15 @@
+  </scm>
   <modules>
     <module>Interface</module>
+	<module>WsdlToDll</module>
```
formatting

anpieber commented on bridge/pom.xml:
```diff
@@ -11,10 +11,15 @@
   </modules>
-</project>
-
+</project>
```
new line at end of file is good

anpieber commented on wsdltodll/pom.xml:
```diff
@@ -0,0 +1,17 @@
+  <artifactId>generate-cs-dll</artifactId>
+  <packaging>maven-plugin</packaging>
+  <version>1.3-SNAPSHOT</version>
+  <name>Convert wsdl files to dll files</name>
```
license, scm, ... is missing (as in other poms too). Though several tags are "standard" for openengsb projects (at least license and scm)

frenzqse replied:
For the scm, shouldn't this one only be defined in the parent pom? Like for example in the bridge? For this package it would be good to make a scm but the other all not? (for example all subpoms in the folder bridge?)

anpieber replied:
easiest is to compare your poms to the poms in the openengsb-framework ;-)

frenzqse replied:
ok. I just saw that every pom has a licence and scm field. But why, when the parent has all the information's? It would be easiert to handle, when there would be a central place.

anpieber replied:
well, a) you've reply scripts and b) it's a (soft) "requirement" from 
oss.sonatype.org

On Tue 07 Feb 2012 02:11:14 PM CET, François Thillen wrote:

> > @@ -0,0 +1,17 @@
> > +<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
> > -  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
> >   +<modelVersion>4.0.0</modelVersion>
> >   +<groupId>org.openengsb.plugin</groupId>
> >   +<artifactId>generate-cs-dll</artifactId>
> >   +<packaging>maven-plugin</packaging>
> >   +<version>1.3-SNAPSHOT</version>
> >   +<name>Convert wsdl files to dll files</name>
> 
> ok. I just saw that every pom has a licence and scm field. But why, when the parent has all the information's? It would be easiert to handle, when there would be a central place.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/openengsb/loom-csharp-common/pull/3/files#r423192

anpieber commented:
in general; pls recheck all license headers

frenzqse committed:
fix all merge problems, formations and added licences

frenzqse committed:
Add licence and scm tag

frenzqse committed:
[OPENENGSB-2722]Change SignalDomain with ExampleDomain.
[OPENENGSB-2727]the plugin is generating a namespace with adding /n:Name to wsdl.exe

frenzqse committed:
[OPENENGSB-2768]Group Method in HelpMethods

frenzqse committed:
[OPENENGSB-2774]Extract the Connector registration Functions
[OPENENGSB-2686]Provide a binary distribution
[OPENENGSB-2497]Start documenting
[OPENENGSB-2775]Extract some Help Method

frenzqse committed:
Add documentation

frenzqse committed:
[OPENENGSB-2497] Documenting the bridge
